ingestion-beam/src/test/java/com/mozilla/telemetry/contextualservices/SendRequestTest.java

```diff
@@ -103,4 +103,35 @@ public void testSuccessfulSend() {
 
     Assert.assertEquals(2, server.getRequestCount());
   }
+
+  @Test
+  public void testRedirectError() {
+    MockWebServer server = new MockWebServer();
+    server.enqueue(new MockResponse().setResponseCode(302));
+
+    Map<String, String> attributes = Collections.singletonMap(Attribute.REPORTING_URL,
+        server.url("/test").toString());
+
+    List<PubsubMessage> input = Collections
+        .singletonList(new PubsubMessage(new byte[0], attributes));
+
+    WithFailures.Result<PCollection<PubsubMessage>, PubsubMessage> result = pipeline
+        .apply(Create.of(input)).apply(SendRequest.of(pipeline.newProvider(true)));
+
+    PAssert.that(result.failures()).satisfies(messages -> {
+      Iterator<PubsubMessage> iterator = messages.iterator();
+      int messageCount = 0;
+
+      for (; iterator.hasNext(); iterator.next()) {
+        messageCount++;
+      }
```

Comment:
```suggestion
      int messageCount = Iterators.size(messages.iterator());
```

Slight preference for using a guava helper in this case.

Reply:
Wow that's useful

Reply:
Guava has a bunch of helpers like this that can make code prettier. Don't expect any magic performance-wise, though. I'm pretty sure `Iterators.size()` is doing exactly the same thing you already have written out here.